src/cephadm/cephadm

```diff
@@ -2929,6 +2972,48 @@ def command_bootstrap():
 
 ##################################
 
+def command_registry_login():
+    if args.registry_json:
+        logger.info(f"Pulling custom registry login info from {args.registry_json}.")
+        with open(args.registry_json) as f:
+            login_info = json.loads(f.read())
+            if "URL" in login_info and "Username" in login_info and "Password" in login_info:
+                args.registry_url = login_info["URL"]
+                args.registry_username = login_info["Username"]
+                args.registry_password = login_info["Password"]
+                logger.info("Logging into custom registry.")
+                try:
+                    out, _, _ = call_throws([container_path, 'login',
+                                               '-u', args.registry_username,
+                                               '-p', args.registry_password,
+                                               args.registry_url])
+                except:
+                    raise Error(f"Failed to login to custom registry @ {args.registry_url} as {args.registry_username} with given password")
+            else:
+                raise Error("json provided for custom registry login did not include all necessary info. "
+                                "Please setup json file as\n"
+                                "{\n"
+                                  " \"URL\": \"REGISTRY_URL\",\n"
+                                  " \"Username\": \"REGISTRY_USERNAME\",\n"
+                                  " \"Password\": \"REGISTRY_PASSWORD\"\n"
+                                "}\n")
```

Comment:
can we merge this code with `get_parm` ?

Reply:
Can you clarify this comment? Is this suggesting I make use of `get_parm` or that I add something to `get_parm`